sympy/combinatorics/rewritingsystem.py

```diff
@@ -127,9 +124,7 @@ def _overlaps(r1, r2):
             len1 = len(r1)
             len2 = len(r2)
             result = []
-            j = 0
-            while j < len1 + len2 - 1:
-                j += 1
+            for j in range(1, len1 + len2 - 1):
```

Comment:
Note that this will make the last `j` equal to `len1 + len2 - 2`. Will that suffice?